tox.ini

```diff
@@ -46,7 +46,7 @@ commands =
     flake8
     black --check --exclude "/(\.eggs|\.git|\.tox|\.venv|\.build|dist|charmhelpers|mod|files)/" .
 deps =
-    black
+    black<24.0.0
```

Comment:
Question: why are we pining `black`? In principal we try to change our code to meet new requirements instead of pining a dependency. But in case it is really necessary, adding a comment would be nice for maintainers and future travellers

Reply:
The tests failed w/o that. But only on the github tests, not locally. @sudeephb asked me to add it. Not sure if that's an issue with this PR, or rather with the github runners...

Reply:
We have been pinning this for a few charms, with the goal of taking care of it later. But I have pushed a new comment by removing the pin and fixing the errors. So it should be good now.